src/components/cylc/tree/TreeItem.vue

```diff
@@ -31,66 +31,82 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
-          :progress="node.node.progress"
-        />
-        <span class="mx-1">{{ node.node.name }}</span>
-      </div>
-      <div :class="getNodeDataClass()" @click="nodeClicked" v-else-if="node.type === 'task-proxy'">
-        <task
-          v-cylc-object="node.node.id"
-          :status="node.node.state"
-          :isHeld="node.node.isHeld"
-          :progress="node.node.progress"
-        />
-        <span class="mx-1">{{ node.node.name }}</span>
-        <div v-if="!isExpanded" class="node-summary">
-          <!-- Task summary -->
-          <job
-            v-for="(task, index) in node.children"
+      <slot name="cyclepoint" v-if="node.type === 'cyclepoint'">
+        <div :class="getNodeDataClass()" @click="nodeClicked">
+          <task
             v-cylc-object="node.node.id"
-            :key="`${task.id}-summary-${index}`"
-            :status="task.node.state"
+            :status="node.node.state"
+            :isHeld="node.node.isHeld"
+            :progress=0
           />
+          <span class="mx-1">{{ node.node.name }}</span>
         </div>
-      </div>
-      <div :class="getNodeDataClass()" v-else-if="node.type === 'job'">
-        <div :class="getNodeDataClass()" @click="jobNodeClicked">
-          <job
+      </slot>
+      <slot name="family-proxy" v-else-if="node.type === 'family-proxy'">
+        <div :class="getNodeDataClass()" @click="nodeClicked">
+          <task
             v-cylc-object="node.node.id"
             :status="node.node.state"
+            :isHeld="node.node.isHeld"
+            :progress="node.node.progress"
           />
-          <span class="mx-1">#{{ node.node.submitNum }}</span>
-          <span class="grey--text">{{ node.node.host }}</span>
+          <span class="mx-1">{{ node.node.name }}</span>
         </div>
-        <!-- leaf node -->
-      </div>
-      <div :class="getNodeDataClass()" v-else>
-        <span @click="nodeClicked" class="mx-1">{{ node.node.name }}</span>
-      </div>
+      </slot>
+      <slot name="task-proxy" v-else-if="node.type === 'task-proxy'">
+        <div :class="getNodeDataClass()" @click="nodeClicked">
+          <task
+            v-cylc-object="node.node.id"
+            :status="node.node.state"
+            :isHeld="node.node.isHeld"
+            :progress="node.node.progress"
+          />
+          <span class="mx-1">{{ node.node.name }}</span>
+          <div v-if="!isExpanded" class="node-summary">
+            <!-- Task summary -->
+            <job
+              v-for="(task, index) in node.children"
+              v-cylc-object="node.node.id"
+              :key="`${task.id}-summary-${index}`"
+              :status="task.node.state" />
+          </div>
+        </div>
+      </slot>
+      <slot name="job" v-else-if="node.type === 'job'">
+        <div :class="getNodeDataClass()">
+          <div :class="getNodeDataClass()" @click="jobNodeClicked">
+            <job
+              v-cylc-object="node.node.id"
```

Comment:
Rebasing it, and almost forgot to include the `v-cylc-object` directives in the cylc objects tags.